src/modules/flow/iio/nodes.c

```diff
@@ -782,12 +782,24 @@ humidity_open(struct sol_flow_node *node, void *data, const struct sol_flow_node
 }
 
 static bool
-adc_create_channels(struct iio_double_data *mdata, int device_id)
+adc_create_channels(struct iio_double_data *mdata, int device_id, const int channel_id)
 {
+    char channel_name[NAME_MAX];
+    int ret;
+
     mdata->iio_base.device = sol_iio_open(device_id, &mdata->iio_base.config);
     SOL_NULL_CHECK(mdata->iio_base.device, false);
 
-    mdata->channel_val = iio_add_channel(mdata->scale, mdata->offset, "in_voltage0", &mdata->iio_base);
+    if (channel_id > 0) {
+        ret = snprintf(channel_name, sizeof(channel_name), "in_voltage%d", channel_id);
+    } else {
+        ret = snprintf(channel_name, sizeof(channel_name), "%s", "in_voltage0");
```

Comment:
Changed the logic, if < 0(without change_id option in .fbp files), will use "in_voltage"